src/devices/Tm1637/Tm1637.cs

```diff
@@ -0,0 +1,305 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+namespace Iot.Device.Tm1637
+{
+    public class Tm1637 : IDisposable
+    {
+        /// <summary>
+        /// The number of segments that the TM1637 can handle
+        /// </summary>
+        public byte MaxSegments => 6;
+        // According to the doc, the clock pulse width minimum is 400 ns
+        // And waiting time between clk up and down is 1 µs
+        private const byte ClockWidthMicroseconds = 1;
+
+        private readonly int _pinClk;
+        private readonly int _pinDio;
+        private GpioController _controller;
+        private byte _brightness;
+        // Default segment order is from 0 to 5
+        private byte[] _segmentOrder = new byte[6] { 0, 1, 2, 3, 4, 5 };
+        // To store what has been displayed last. Used when change on brightness or
+        // screen on/off is used
+        private byte[] _lastDisplay = new byte[6] { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+        private bool _secrrenOn;
+
+        /// <summary>
+        /// Initialize a TM1637
+        /// </summary>
+        /// <param name="pinClk">The clock pin</param>
+        /// <param name="pinDio">The data pin</param>
+        /// <param name="pinNumberingScheme">Use the logical or physical pin layout</param>
+        public Tm1637(int pinClk, int pinDio, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical)
+        {
+            _pinClk = pinClk;
+            _pinDio = pinDio;
+            _controller = new GpioController(pinNumberingScheme);
+            _controller.OpenPin(_pinClk, PinMode.Output);
+            _controller.OpenPin(_pinDio, PinMode.Output);
+            _brightness = 7;
+        }
+
+        /// <summary>
+        /// Order of segments, expect a 6 length byte array
+        /// 0 to 5, any order. Most of the time 4 segments do
+        /// not need to be changed but the 6 ones may be in different order
+        /// like 0 1 2 5 4 3. In this case, this byte array has be be in this order
+        /// </summary>
+        public byte[] SegmentOrder
+        {
+            get { return _segmentOrder; }
+            set
+            {
+                if (value.Length != MaxSegments)
+                    throw new ArgumentException($"Size of {nameof(SegmentOrder)} can only be 6 length");
+                // Check if we have all values from 0 to 5
+                bool allExist = true;
+                for (int i = 0; i < MaxSegments; i++)
+                {
+                    allExist &= Array.Exists(value, e => e == i);
+                }
+                if (!allExist)
+                    throw new ArgumentException($"{nameof(SegmentOrder)} needs to have all existing segments from 0 to 5");
+                _segmentOrder = value;
```

Comment:
this should copy, otherwise changing the array after setting will affect how this device works